model/pdata/common.go

```diff
@@ -795,8 +795,25 @@ func (am AttributeMap) CopyTo(dest AttributeMap) {
 	*dest.orig = origs
 }
 
+// AsMap converts an OTLP AttributeMap to a standard go map
+func (am AttributeMap) AsMap() map[string]AttributeValue {
+	rawMap := make(map[string]AttributeValue)
+	am.Range(func(k string, v AttributeValue) bool {
+		rawMap[k] = v
+		return true
+	})
+	return rawMap
+}
+
 // AttributeMapToMap converts an OTLP AttributeMap to a standard go map
+// Deprecated: use AttributeMap's AsMap method instead.
 func AttributeMapToMap(attrMap AttributeMap) map[string]interface{} {
+	return attributeMapToMarshalable(attrMap)
+}
```

Comment:
So what I am hearing is that it is still useful to have "map[string]interface{}" for JSON encoding. Then let's do the following:
1. Have a `func (am AttributeMap) AsMap() map[string]AttributeValue`
2. Have a `func (am AttributeMap) AsRaw() map[string]interface{}`

We can then have an equivalent `AsRaw` on AttributeValue and on ArrayAny that can be used. 

In this way we follow the pattern that `AsRaw` returns "raw" structs.

Reply:
Implemented `AsRaw`, PTAL